### PR TITLE
Properly populate the FreeRTOS type sizes and offsets.

### PR DESCRIPTION
**Pull request by Du-Chao**
1. According to FreeRTOS/portable/GCC/RISC-V/portmacro.h, align _TickType_t_ with UBASE_TYPE.
2. Make the alignment of _ListItem_t_ clear.

### CODE REVIEW

File: src/rtos/FreeRTOS.c [hunk 1]
```diff
@@ -352,6 +353,12 @@ static void freertos_compute_offsets(struct rtos *rtos)
 
 	freertos->pointer_size = DIV_ROUND_UP(target_address_bits(rtos->target), 8);
 	freertos->ubasetype_size = DIV_ROUND_UP(target_data_bits(rtos->target), 8);
+	/*
+	 * According to FreeRTOS/portable/GCC/RISC-V/portmacro.h,
+	 * the TickType_t is defined as UBASE_TYPE,
+	 * Not controlled by configUSE_16_BIT_TICKS.
+	 */
+	freertos->ticktype_size = freertos->ubasetype_size;
```
Comment (timsifive) [inline src/rtos/FreeRTOS.c:361]: If ticktype_size is always set to ubasetype_size, what's the benefit of having ticktype_size defined in the struct?

File: src/rtos/FreeRTOS.c [hunk 2]
```diff
@@ -352,6 +353,12 @@ static void freertos_compute_offsets(struct rtos *rtos)
 
 	freertos->pointer_size = DIV_ROUND_UP(target_address_bits(rtos->target), 8);
 	freertos->ubasetype_size = DIV_ROUND_UP(target_data_bits(rtos->target), 8);
+	/*
+	 * According to FreeRTOS/portable/GCC/RISC-V/portmacro.h,
```
Comment (timsifive) [inline src/rtos/FreeRTOS.c:357]: What about other ports? This code should work on non-RISC-V targets as well.

File: src/rtos/FreeRTOS.c (hunk 1 above)
Comment (Du-Chao) [reply src/rtos/FreeRTOS.c:361]: The benefit is to make the calculation more clear and comprehensible.

File: src/rtos/FreeRTOS.c (hunk 2 above)
Comment (Du-Chao) [reply src/rtos/FreeRTOS.c:357]: I thought that _riscv-openocd_ is only for RISC-V port, so this PR only considered the RISC-V case.

**Comment by Du-Chao** [comment]
Thanks Tim for the comment.

I will close this PR and then try to prepare another patch which is suitable for all targets.

File: src/rtos/FreeRTOS.c (hunk 2 above)
Comment (TommyMurphyTM1234) [reply src/rtos/FreeRTOS.c:357]: > I thought that _riscv-openocd_ is only for RISC-V port, so this PR only considered the RISC-V case.

RISC-V specific changes, in the `src/target/riscv` folder and above (the latter potentially affecting other/all target architectures) in this repo, are eventually upstreamed to the master OpenOCD project.

Comment (Du-Chao) [reply src/rtos/FreeRTOS.c:357]: > 

I see. Thanks for the explanation.
BTW, I created another PR #806 which may benefit all targets.

Comment (Du-Chao) [reply src/rtos/FreeRTOS.c:357]: > > I thought that _riscv-openocd_ is only for RISC-V port, so this PR only considered the RISC-V case.
> 
> RISC-V specific changes, in the `src/target/riscv` folder and above (the latter potentially affecting other/all target architectures) in this repo, are eventually upstreamed to the master OpenOCD project.

I see. Thanks for the explanation.
BTW, I created another PR https://github.com/riscv/riscv-openocd/pull/806 which may benefit all targets.

**Pull request closed without merging**